File: binarySearch/bs.py
#Program to find a number by user in a list using binary search

def binary_search(num, num_list):
    u_limit = len(num_list) - 1
    l_limit = 0
    stop = 0

    while(stop != 1):
        guess = int((u_limit+l_limit)/2)

        if num_list[guess] > num:
            u_limit = guess - 1
        elif num_list[guess] < num:
            l_limit = guess + 1
        else:
            stop = 1
            print("The number is in the list. Index : {}".format(guess))

        if l_limit > u_limit:
            stop = 1
            print("The number is not in the list.")

prime_num = [2, 3, 5, 7, 11, 13, 17, 19, 23, 29, 31, 37, 41, 43, 47, 53, 59, 61, 67, 71, 73, 79, 83, 89, 97]
find_num = int(input("Please enter a prime number from 1 to 97 : "))

binary_search(find_num, prime_num)
    

File: binarySearch/bs.c
#include <stdio.h>

//This program accepts a user input and checks if it is within an ordered list.

int main(void){
	int num_list[]={2, 3, 5, 7, 11, 13, 17, 19, 23, 29, 31, 37, 41, 43, 47, 53, 59, 61, 67, 71, 73, 79, 83, 89, 97}, find, stop, guess, min ,max;
	// int i,j;

	// srand(time(NULL)); 
	/*time(NULL) is the time now, which is the number of seconds from epoch. 
	Since number of seconds changes with time, seed is unqiue, hence the rand() can produce more random numbers*/
	// answer = (rand() % 100) + 1;
	printf("Enter a prime number from 1 to 97 : ");
	scanf("%d", &find);
	/*random number from 1 to 100. Taking the remainder works because when the random number is divided by 100, there is no way the remainder can be less than 0 or more than 99.
	So to get 1 to 100, just add 1.*/

	// for(j = 0; j < 100; j++){ //To print the array
	// 	printf("%d ",num_list[j]);
	// }

	min = 0;
	max = sizeof(num_list)/sizeof(num_list[0]);

	do{
		guess = (min+max)/2;
			printf("%d\n", num_list[guess]);
			if(num_list[guess] > find){
				max = guess - 1;
				printf("max : %d\n", max);
			} else if(num_list[guess] < find) {
				min = guess + 1;
				printf("min : %d\n", min);
			} else {
				stop = 1;
				printf("The number in the the list. Index : %d", guess);
			}

			//Terminate if cannot find
			if(min > max){
				stop = 1;
				printf("The number is not found.");
			}

	}while(stop != 1);
	return 0;
}
